code/modules/vehicles/armored/armored_actions.dm

```diff
@@ -0,0 +1,128 @@
+#define ARMOR_PASSENGER "Passenger"
+
+/datum/action/vehicle/sealed/armored/swap_seat/action_activate(trigger_flags)
+	if(!transfer_checks())
+		return
+	var/list/choices = list()
+	if(!chassis.is_driver(owner))
+		choices += ARMOR_DRIVER
+	if(!chassis.is_equipment_controller(owner))
+		choices += ARMOR_GUNNER
+	choices += ARMOR_PASSENGER
+	var/choice = tgui_input_list(owner, "Select a seat", chassis.name, choices)
+	if(!choice)
+		return
+	if(!transfer_checks(choice))
+		return
+	chassis.balloon_alert(owner, "moving to other seat...")
+	if(!do_after(owner, chassis.enter_delay, target = chassis, extra_checks=CALLBACK(src, PROC_REF(transfer_checks), choice)))
+		chassis.balloon_alert(owner, "interrupted!")
+		return
+	chassis.remove_control_flags(owner, VEHICLE_CONTROL_MELEE|VEHICLE_CONTROL_EQUIPMENT|VEHICLE_CONTROL_DRIVE|VEHICLE_CONTROL_SETTINGS)
+	switch(choice)
+		if(ARMOR_GUNNER)
+			chassis.balloon_alert(owner, "controlling gunner seat")
+			chassis.add_control_flags(owner, VEHICLE_CONTROL_MELEE|VEHICLE_CONTROL_EQUIPMENT)
+		if(ARMOR_DRIVER)
+			chassis.balloon_alert(owner, "controlling pilot seat")
+			chassis.add_control_flags(owner, VEHICLE_CONTROL_DRIVE|VEHICLE_CONTROL_SETTINGS)
+		if(ARMOR_PASSENGER)
+			chassis.balloon_alert(owner, "entered passenger seat")
+
+///checks if owner can still transfer
+/datum/action/vehicle/sealed/armored/swap_seat/proc/transfer_checks(choice)
+	if(!owner || !chassis || !(owner in chassis.occupants))
+		return FALSE
+	if(length(chassis.occupants) >= chassis.max_occupants)
+		chassis.balloon_alert(owner, "other seats occupied!")
+		return FALSE
+	switch(choice)
+		if(ARMOR_GUNNER)
+			if(length(chassis.return_controllers_with_flag(VEHICLE_CONTROL_EQUIPMENT)) >= 1)
+				chassis.balloon_alert(owner, "gunner occupied!")
+				return FALSE
+		if(ARMOR_DRIVER)
+			if(chassis.driver_amount() >= chassis.max_drivers)
+				chassis.balloon_alert(owner, "driver occupied!")
+				return FALSE
+	return TRUE
+
+#undef ARMOR_DRIVER
+#undef ARMOR_GUNNER
+#undef ARMOR_PASSENGER
+
+/datum/action/vehicle/sealed/armored/toggle_lights
+	name = "Toggle Lights"
+	action_icon_state = "mech_lights_off"
+
+/datum/action/vehicle/sealed/armored/toggle_lights/action_activate(trigger_flags)
+	if(!owner || !chassis || !(owner in chassis.occupants))
+		return
+
+	if(!(chassis.flags_armored & ARMORED_HAS_HEADLIGHTS))
+		chassis.balloon_alert(owner, "the vehicle's lights are broken!")
+		return
+	chassis.flags_armored ^= ARMORED_LIGHTS_ON
+	if(chassis.flags_armored & ARMORED_LIGHTS_ON)
+		action_icon_state = "mech_lights_on"
+	else
+		action_icon_state = "mech_lights_off"
+	chassis.set_light_on(chassis.flags_armored & ARMORED_LIGHTS_ON)
+	chassis.balloon_alert(owner, "toggled lights [chassis.flags_armored & ARMORED_LIGHTS_ON ? "on":"off"]")
+	playsound(chassis,'sound/mecha/brass_skewer.ogg', 40, TRUE)
+	chassis.log_message("Toggled lights [(chassis.flags_armored & ARMORED_LIGHTS_ON)?"on":"off"].", LOG_MECHA)
+	update_button_icon()
```

Comment:
set_toggle() will already update the button, unless you're bypassing it somehow.

Reply:
this doesnt use set toggled, its toggling something on the vehicle obj not on the action icon